internal/daemon/api_signals.go

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2021 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package daemon
+
+import (
+	"encoding/json"
+	"net/http"
+)
+
+type signalsPayload struct {
+	Signal   string   `json:"signal"`
+	Services []string `json:"services"`
+}
+
+func v1PostSignals(c *Command, req *http.Request, _ *userState) Response {
+	var payload signalsPayload
+	decoder := json.NewDecoder(req.Body)
+	if err := decoder.Decode(&payload); err != nil {
+		return statusBadRequest("cannot decode request body: %v", err)
+	}
+	if len(payload.Services) <= 0 {
```

Comment:
In other APIs do we not deal with the wildcard case (might just be autostart?)? Here it probably doesn't make sense just wondering if we should seek consistency.

Reply:
Yeah, `replan` and `autostart` error if you *do* provide service names, and `start`, `stop`, and `restart` error if you don't (which makes sense for each of those, I think). I don't think it's a good idea (or makes a lot of sense) to send a signal to all services. If someone really wants to they can do something like this:

```python
service_names = [s.name for s in container.get_services()]
container.send_signal('SIGHUP', service_names)
```